Lib/json/tool.py

```diff
@@ -11,9 +11,28 @@
 
 """
 import argparse
+import contextlib
 import json
+import os.path
 import sys
-from pathlib import Path
+import tempfile
+
+
+@contextlib.contextmanager
+def _staged_outfile(path):
+    # Use tempfile to support outfile == infile case.
+    # See https://discuss.python.org/t/adding-atomicwrite-in-stdlib/11899
+    dir, base = os.path.split(path)
+    fd, tempname = tempfile.mkstemp(prefix=base + '.', dir=dir)
+    file = os.fdopen(fd, "w", encoding="utf-8")
+    try:
+        with file:
+            yield file
+    except BaseException as e:
```

Comment:
```suggestion
    except:
```

Reply:
I used BaseException because flake8 reports error for `except:`.
But there are many flake8 errors in json. I decided to ignore it.